powersoftau/src/batched_accumulator.rs

```diff
@@ -149,106 +139,94 @@ impl<'a, E: Engine + Sync> BatchedAccumulator<'a, E> {
+                return Err(VerificationError::InvalidGenerator(ElementType::TauG1).into());
             }
             if after.tau_powers_g2[0] != E::G2Affine::prime_subgroup_generator() {
-                error!("tau_powers_g2[0] != 1");
-                return false;
-            }
-
-            // Did the participant multiply the previous tau by the new one?
-            if !same_ratio(
-                (before.tau_powers_g1[1], after.tau_powers_g1[1]),
-                (tau_g2_s, key.tau_g2),
-            ) {
-                error!("Invalid ratio (before.tau_powers_g1[1], after.tau_powers_g1[1]), (tau_g2_s, key.tau_g2)");
-                return false;
-            }
-
-            // Did the participant multiply the previous alpha by the new one?
-            if !same_ratio(
-                (before.alpha_tau_powers_g1[0], after.alpha_tau_powers_g1[0]),
-                (alpha_g2_s, key.alpha_g2),
-            ) {
-                error!("Invalid ratio (before.alpha_tau_powers_g1[0], after.alpha_tau_powers_g1[0]), (alpha_g2_s, key.alpha_g2)");
-                return false;
+                return Err(VerificationError::InvalidGenerator(ElementType::TauG2).into());
             }
 
-            // Did the participant multiply the previous beta by the new one?
-            if !same_ratio(
-                (before.beta_tau_powers_g1[0], after.beta_tau_powers_g1[0]),
-                (beta_g2_s, key.beta_g2),
-            ) {
-                error!("Invalid ratio (before.beta_tau_powers_g1[0], after.beta_tau_powers_g1[0]), (beta_g2_s, key.beta_g2)");
-                return false;
-            }
-            if !same_ratio(
-                (before.beta_tau_powers_g1[0], after.beta_tau_powers_g1[0]),
-                (before.beta_g2, after.beta_g2),
-            ) {
-                error!("Invalid ratio (before.beta_tau_powers_g1[0], after.beta_tau_powers_g1[0]), (before.beta_g2, after.beta_g2)");
-                return false;
+            let check_ratios = &[
+                // Did the participant multiply the previous tau by the new one?
+                (
+                    (before.tau_powers_g1[1], after.tau_powers_g1[1]),
+                    tau_g2_check,
+                    "Before-After: Tau[1] G1<>G2",
+                ),
+                // Did the participant multiply the previous alpha by the new one?
+                (
+                    (before.alpha_tau_powers_g1[0], after.alpha_tau_powers_g1[0]),
+                    alpha_g2_check,
+                    "Before-After: Alpha[0] G1<>G2",
+                ),
+                // Did the participant multiply the previous beta by the new one?
+                (
+                    (before.beta_tau_powers_g1[0], after.beta_tau_powers_g1[0]),
+                    beta_g2_check,
+                    "Before-After: Beta[0] G1<>G2",
+                ),
+                // todo: since we're checking with the same G1 elements above, can't we remove the same_ratio
```

Comment:
As discussed, I think this could be a redundant call which can be replaced with a check that `(before.beta_g2, after.beta_g2) == beta_g2_check`